Tests/RouterTests.swift

```diff
@@ -43,14 +43,55 @@ private final class RouterTestServer: NSURLProtocol {
     override func startLoading() {
         guard let requestURL = request.URL,
                   client = client else { return }
-        
+
         client.URLProtocol(self, didReceiveResponse: NSHTTPURLResponse(URL: requestURL, statusCode: 200, HTTPVersion: "HTTP/1.1", headerFields: nil)!, cacheStoragePolicy: .AllowedInMemoryOnly)
         client.URLProtocolDidFinishLoading(self)
     }
 
     override func stopLoading() {}
 }
 
+private final class ProtocolClientTest: NSObject, NSURLProtocolClient {
+
+    private let shouldFail_URLProtocolDidFinishLoading: Bool
+    
+    var didCall_URLProtocolDidFinishLoading: Bool = false
+    
+    init(shouldFail_URLProtocolDidFinishLoading: Bool) {
+        self.shouldFail_URLProtocolDidFinishLoading = shouldFail_URLProtocolDidFinishLoading
+        super.init()
+    }
+
+    @objc func URLProtocol(`protocol`: NSURLProtocol, wasRedirectedToRequest request: NSURLRequest, redirectResponse: NSURLResponse) {
+        // intentionally left empty
+    }
+    @objc func URLProtocol(`protocol`: NSURLProtocol, cachedResponseIsValid cachedResponse: NSCachedURLResponse) {
+        // intentionally left empty
+    }
+    @objc func URLProtocol(`protocol`: NSURLProtocol, didReceiveResponse response: NSURLResponse, cacheStoragePolicy policy: NSURLCacheStoragePolicy) {
+        // intentionally left empty
+    }
+    @objc func URLProtocol(`protocol`: NSURLProtocol, didLoadData data: NSData) {
+        // intentionally left empty
+    }
+    @objc func URLProtocolDidFinishLoading(`protocol`: NSURLProtocol) {
+        if shouldFail_URLProtocolDidFinishLoading {
+            XCTFail("Protocol client should never get 'URLProtocolDidFinishLoading' called.")
+        }
+        
+        didCall_URLProtocolDidFinishLoading = true
+    }
+    @objc func URLProtocol(`protocol`: NSURLProtocol, didFailWithError error: NSError) {
+        // intentionally left empty
+    }
+    @objc func URLProtocol(`protocol`: NSURLProtocol, didReceiveAuthenticationChallenge challenge: NSURLAuthenticationChallenge) {
+        // intentionally left empty
+    }
+    @objc func URLProtocol(`protocol`: NSURLProtocol, didCancelAuthenticationChallenge challenge: NSURLAuthenticationChallenge) {
+        // intentionally left empty
+    }
+}
+
```

Comment:
I need the `ProtocolClientTest` to manually simulate the control flow in the `KakapoServer` (see: `"should send notifications when loading has finished"`).

Because there's no way for me to access the `KakapoServer` instances, which are created automatically be the OS. Thus I create one `KakapoServer` instance manually to test drive the `start` & `stop` logic.